library/bignum_core.c

```diff
@@ -824,8 +856,8 @@ static void mbedtls_mpi_core_exp_mod_optionally_safe(mbedtls_mpi_uint *X,
     /* We'll process the bits of E from most significant
      * (limb_index=E_limbs-1, E_bit_index=biL-1) to least significant
      * (limb_index=0, E_bit_index=0). */
-    size_t E_limb_index = E_limbs;
-    size_t E_bit_index = 0;
+    size_t E_limb_index;
+    size_t E_bit_index;
```

Comment:
Non-blocker: I don't like relying on auxiliary functions to initialize variables. Unfortunately compilers aren't good at analyzing whether variables are analyzed on all code paths, and we don't run Coverity assiduously enough to notice all problems. I would strongly prefer to initialize to the safe-path values here, and have `exp_mod_calc_first_bit_optionally_safe` modify the values on the unsafe path.